images/scripts/make-image-tag.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright 2017-2020 Authors of Cilium
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# This script provides two image tagging mechanisms.
+#
+# For general images that use most of the tree as input, it's most sensible to
+# use git commit hash as a tag.
+#
+# For images that use contents of a subdirectory as input, it's convenient to use
+# a git tree hash. Running `git show` with tree hash based tag will display the
+# contents of the subdirectory that was used as build input, mitigating any doubts
+# in what was used to build this image.
+#
+# For both types of tags To differentiate any non-authoritative builds, i.e.
+# builds from development branches, `-dev` suffix is added. Any builds that may
+# include uncommitted changes will have `-wip` tag.
+
+# TODO: determine when the head is also a release tag; that it can be done with
+# `git name-rev --name-only --tags --no-undefined HEAD` however Cilium usually
+# gets two tags one with `v` prefix and one without, beacause of that the output
+# of `git name-rev` maybe one or the other; to implement that with certainty
+# release process/automation needs to be reviewed in detail
+
+if [ "$#" -gt 1 ] ; then
+  echo "$0 supports exactly 1 or no arguments"
```

Comment:
```suggestion
  echo "usage: $0 [IMAGE_DIR]"
```

Reply:
Maybe, I'm just keeping these simple for now. In the future, I'd rather turn these into a Go program actually (see https://github.com/cilium/image-tools/issues/10). I think that way we could enforce best practice and avoid all the boilerplate and manual steps.